atomix/cluster/src/main/java/io/atomix/raft/roles/PassiveRole.java

```diff
@@ -186,7 +188,18 @@ private void addSnapshotListener() {
     }
 
     final var snapshotChunk = new SnapshotChunkImpl();
-    snapshotChunk.wrap(new UnsafeBuffer(request.data()), 0, request.data().capacity());
+    try {
+      snapshotChunk.wrap(new UnsafeBuffer(request.data()), 0, request.data().capacity());
```

Comment:
Would be good to check if it is the right type by "tryWrap".

Reply:
Try wrap checks that the version is also the same, but I'm thinking this is actually wrong behaviour, no? Handling the different versions should be done depending on the messages, so we could use `tryWrap` and simply remove the version check. wdyt?

Reply:
`tryWrap` only checks schemaId and templateId, right? It doesn't check the version.

Reply:
Ah, correct, not sure why I was so sure it did :sweat_smile:

Reply:
The one downside I see now is that using `tryWrap`, you don't know what the error that occurred was (without breaking the abstraction). But I also like that it doesn't throw an exception... :thinking: